include/gz/physics/Joint.hh

```diff
@@ -584,6 +586,46 @@ namespace gz
       };
     };
 
+    /////////////////////////////////////////////////
+    /// \brief This feature sets the mimic constraint for this Joint.
+    class GZ_PHYSICS_VISIBLE SetMimicConstraintFeature
+        : public virtual Feature
+    {
+      /// \brief The Joint API for setting the mimic joint constraint.
+      /// \param[in] _joint
+      ///   name of the joint to be mimicked, i.e. the leader joint.
+      /// \param[in] _multiplier
+      ///   The multiplier to be applied to poistion of leader joint.
+      /// \param[in] _offset
+      ///   The offset to be applied to position of leader joint after
+      ///   the multiplier is applied.
+      public: template <typename PolicyT, typename FeaturesT>
+      class Joint : public virtual Feature::Joint<PolicyT, FeaturesT>
+      {
+        public: using Scalar = typename PolicyT::Scalar;
+
+        public: void SetMimicConstraint(
```

Comment:
we should add a `std::size_t _dof` argument  (similar to the [GetBasicJointState APIs](https://github.com/gazebosim/gz-physics/blob/gz-physics6/include/gz/physics/Joint.hh#L45-L73)) to specify the follower joint axis

Reply:
https://github.com/gazebosim/gz-physics/pull/431/commits/d7f552fe0ff38086eada8fd62b435eae1bb347c8